GVFS/GVFS.Common/GVFSLock.Shared.cs

```diff
@@ -96,7 +97,7 @@ public partial class GVFSLock
             }
 
             result = null;
-            return true;
+            return isSuccessfulLockResult;

```

Comment:
Thanks for the investigations. Yea this is a pretty glaring hole. We should definitely follow up to find out why we didn't have a reliably failing test as a result of this bug.

Reply:
Reading through this, and it's not as bad as it seems - which also explains the lack of test failures. In the normal case, this method blocks until the lock is acquired, so the return value is always true. However, that assumption fails once an unmount is initiated, in which case we have to pass through the false result. Constructing a reliable test for that will be tricky, since it's so timing dependent. Your fix does make sense, and assuming that the git status cache stops misbehaving during unmount, that is a pretty good validation of this fix too.